### DISCUSSION TITLE
Mac上chrome系列浏览器的奇怪问题以及解法

### DISCUSSION THREAD

**Discussion by guyschaos** [kkkgo/PaoPaoGateWay#111] [问和答]
先称赞一下项目作者。这个项目很棒，简单精巧。我已经从原来的自行搭建透明代理转换为ppgw。其中我觉得最值得称道的点是

- 以ISO形式发布，用最瘦的虚拟机的形式提供网关
- 轮询更新配置，免去更新后需要重启虚拟机的麻烦
- 以dns->fake-ip->静态路由的传递关系来控制网络走向，改变了我之前以dhcp option 3/6 的方式下需要重走dhcp分配地址才能切换网络环境的麻烦

总得来说，我觉得作者网络功力深厚，看回复也很仔细。希望项目会越来越好。(私以为让ISO保持简单是个更好的选择，尽量控制在ISO虚拟机中的所实现的协议，也不要贸然切到sing-box)

--------------

接着我来描述一下我碰到的问题

## 网络环境

- 路由器 Openwrt x86 
- Proxy 群晖Docker 
- AP Redmi AX6000

## 配置

SmartDNS 作为DNS入口， SmartDNS 地址 192.168.2.210 也作为整个局域网的DNS server，由Openwrt 分配给网络内所有设备
SmartDNS会根据请求者ip/mac分配到不同的上游DNS服务器（也就是说，我可以通过静态分配ip地址或者设定mac地址到smartdns来控制设备网络走向）

- cn -> smartdns -> adguard cn -> public dns (ali dns/114)
- global -> smartdns -> adguard global -> ppdns

## 问题

chrome/opera/vivaldi这些浏览器在上文的global环境下，访问非国内网站从开发者工具里看有大量的connection reset/close日志，看不到请求地址，经常无法加载网页。

同时：

- waterfox可以正常浏览
- nslookup是正常的fake ip
- curl可以正常连接网页
- 另一台windows上所有浏览器都正常
- 手机也是正常

## 解法

翻文档，设置dhcp option 121后网络正常

## 原因？

我猜测是Chrome核心的某个特性造成的这个问题？这个特性甚至能超越外部的路由表，把数据往它认为该去的地方送？ 除非Mac本机上有特定的路由(option 121指派的路由)它才去遵照这个路由？
知道的大哥提个醒吧，想知道真正的原因。

**Comment by mouyase** [comment]
chrome关闭使用安全dns了吗

**Reply by guyschaos** [reply]
关闭的。在smartdns和adguard的log中都是看到dns请求日志的。

**Reply by UpwardRun** [reply]
在Windows Chrome中我也出现了这种情况

**Reply by guyschaos** [reply]
@UpwardRun 最后是怎么解决的？

**Reply by UpwardRun** [reply]
最开始把paopaodns重新安装+gateway重新制作了一次，然后重新安装了一下，发现好了，然后后面改成了paopaodns+upcg+singbox实现透明代理把gateway替换掉了

**Comment by kkkgo** [comment]
如果单纯是curl可以但chrome之类不行的话可能是http/2或者http/3导致的（跟节点协议支持或者内核可能也有关系，可以docker定制sniff=yes尝试），浏览器安全DNS也是一种情况（但很好排查，你只要访问一个没被墙的国外地址），但你加dhcp 121就又可以了的话就确实无法解释了（也许是偶然）。
